### ISSUE TITLE
Make the console an optional install

### ISSUE THREAD

**Issue by jni** [napari/napari#1593]
## 🚀 Feature

It's not great that installing napari brings down the entire jupyter stack because of the console. We should remove the button both in IPython (as already discussed) and when the QtConsole dependencies are not there. `pip install napari[all]` would bring down the console stack, as would `pip install napari[pyqt5,qtconsole]`. 

## Motivation

We should aim to make napari a lean viewer. Originally, we made the console standard by default because we were concerned about having a consistent napari experience everywhere. That point will be moot with plugins and extensions, and anyway the button doesn't work on IPython, so I think it's about the right time to make this move.

Eventually, the console itself could be an (externally installed) extension.

**Comment by tlambert03** [comment]
> Eventually, the console itself could be an (externally installed) extension.

👍... it already pretty closely matches the description of the proposed dock widget extensions in #1327

**Comment by sofroniewn** [comment]
> 👍... it already pretty closely matches the description of the proposed dock widget extensions in #1327

agreed, this would be a good test case for that hookspec, and at that point would could have a new repo `napari-qtconsole` (or something similar) which would just be a plugin, which would be pretty cool!

**Comment by sofroniewn** [comment]
Probably first step is to get #1327 (or something similar) in, possibly under an "experimental" namespace (see discussion in #1564), and then follow up with this refactor. Thoughts @jni @tlambert03?

**Comment by jni** [comment]
That's definitely the medium term vision, but I am proposing that we can make it optional even before that.

**Comment by sofroniewn** [comment]
With the merge of #2118 it should now be relatively easy to do a followups

1. Make the console optional
2. Remove the button if the console is not present or if launching from ipython
3. make the console use the dock widget hookspec
